Bearded.Utilities.Tests/Collections/DeletableObjectListTests.cs

```diff
@@ -473,6 +473,106 @@ public void WorksNonGenerically()
                 enumeratedItems.Should().BeEquivalentTo(items, withExactSameItems);
             }
         }
+        
+        public class ReverseEnumeration
+        {
+            [Property]
+            public void SkipsDeletedItems(NonEmptyArray<bool> someBools)
+            {
+                var boolQueue = someBools.Get.Looping();
+                var (list, items) = createPopulatedList(20);
+
+                foreach (var item in list)
+                {
+                    if (boolQueue.Next())
+                        item.Deleted = true;
+                }
+
+                list.Reversed.Should().BeEquivalentTo(items.Reverse().Where(i => !i.Deleted), withExactSameItems);
+            }
+
+            [Fact]
+            public void SkipsItemsDeletedAheadOfEnumeration()
+            {
+                var (list, items) = createPopulatedList(20);
+                var enumeratedItems = new List<TestDeletable>();
+
+                foreach (var item in list.Reversed)
+                {
+                    enumeratedItems.Add(item);
+                    list.First().Deleted = true;
+                }
+
+                enumeratedItems.Should().BeEquivalentTo(items.Reverse().Take(10), withExactSameItems);
+            }
+            
+            [Fact]
+            public void SkipsItemsAddedDuringEnumeration()
+            {
+                var (list, items) = createPopulatedList(20);
+                var enumeratedItems = new List<TestDeletable>();
+
+                foreach (var item in list.Reversed)
+                {
+                    enumeratedItems.Add(item);
+                    list.Add(new TestDeletable());
+                }
+
+                enumeratedItems.Should().BeEquivalentTo(items.Reverse(), withExactSameItems);
+            }
+
+            [Fact]
+            public void IsNotAffectedByItemsBeingDeletedAfterEnumeratingThem()
+            {
+                var (list, items) = createPopulatedList(20);
+                var enumeratedItems = new List<TestDeletable>();
+                
+                foreach (var item in list.Reversed)
+                {
+                    enumeratedItems.Add(item);
+                    list.Last().Deleted = true;
+                }
+                
+                enumeratedItems.Should().BeEquivalentTo(items.Reverse(), withExactSameItems);
+                list.Should().BeEmpty();
+            }
+
+            [Fact]
+            public void CanBeResetTakingIntoAccountDeletedAndRemovedItems()
+            {
+                var (list, items) = createPopulatedList(20);
+                var enumeratedItems = new List<TestDeletable>();
+
+                // ReSharper disable once GenericEnumeratorNotDisposed
+                var enumerator = list.Reversed.GetEnumerator();
+                enumerator.MoveNext();
+                enumerator.MoveNext();
+                enumerator.MoveNext();
+                
+                list.Remove(items[19]);
```

Comment:
You are testing two different behaviours in a single test. Tests should generally only test a single behaviour each, to make it clear what is broken when a test fails.